components/dashboard/src/service/service.tsx

```diff
@@ -28,18 +29,40 @@ import {
     watchWorkspaceStatusInOrder,
 } from "../data/workspaces/listen-to-workspace-ws-messages2";
 import { Workspace, WorkspaceSpec_WorkspaceType, WorkspaceStatus } from "@gitpod/public-api/lib/gitpod/v1/workspace_pb";
-import { sendTrackEvent } from "../Analytics";
+import { sendTrackEvent, trackEvent } from "../Analytics";
+import { getFeatureFlagValue } from "../experiments/flags";
+import UAParser from "ua-parser-js";
 
 export const gitpodHostUrl = new GitpodHostUrl(window.location.toString());
 
-function createGitpodService<C extends GitpodClient, S extends GitpodServer>() {
-    let host = gitpodHostUrl.asWebsocket().with({ pathname: GitpodServerPath }).withApi();
+const gitpodHost = gitpodHostUrl.asWebsocket().with({ pathname: GitpodServerPath }).withApi().toString();
 
+const isFirefox = new UAParser().getBrowser().name?.toLowerCase().includes("firefox") ?? false;
```

Comment:
fyi https://github.com/microsoft/vscode/blob/2e3a1d8bd9270480fc548d9351c7efebfd52bfed/src/vs/base/browser/browser.ts#L101-L103

Reply:
The way we're using now is a copy from browser extension advertisement https://github.com/gitpod-io/gitpod/blob/bf27597ae5f69345ddb8a68727b74b6ac7c7c362/components/dashboard/src/workspaces/BrowserExtensionBanner.tsx#L109-L110